android/quest/src/main/java/org/smartregister/fhircore/quest/util/IterableUtil.kt

```diff
@@ -28,3 +31,26 @@ package org.smartregister.fhircore.quest.util
 fun <A, B> nonNullGetOrDefault(map: Map<A, B>, key: A?, defaultValue: B): B {
   return if (key != null) map.getOrDefault(key, defaultValue) else defaultValue
 }
+
+/**
+ * Function to convert the elements of an array that have paramType PARAM_DATA to a map of their
+ * keys to values.
+ *
+ * @property array The array of ActionParameter elements to convert
+ *
+ * @return Map of the values or emptyMap if [array] is null
+ */
+fun convertActionParameterArrayToMap(array: Array<ActionParameter>?): Map<String, String> {
+  val paramsMap: MutableMap<String, String> = mutableMapOf()
+  if (array != null) {
+    for (param in array) {
+      if ((param.paramType == ActionParameterType.PARAMDATA) ||
+          (param.paramType == ActionParameterType.UPDATEABLEIDS) && !param.value.isNullOrEmpty()
+      ) {
+        paramsMap[param.key] = param.value
+      }
+    }
+    return paramsMap.toMap()
+  }
+  return emptyMap()
+}
```

Comment:
This looks like it should extend this function, https://github.com/opensrp/fhircore/blob/main/android/quest/src/main/java/org/smartregister/fhircore/quest/util/extensions/ConfigExtensions.kt#L140, not be added here

```suggestion
```

Reply:
in that extension you can use

```kotlin
param.paramType in listOf(ActionParameterType.PARAMDATA, ..)
```

to reduce duplication